clamav-mirror/test.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+# 
+# Script to test and publish definitions
+set -o errexit
+
+CVDUPDATE_DEST="${CVDUPDATE_DEST:-/home/clam/db}"
+# CVDUPDATE_SLEEP="${CVDUPDATE_SLEEP:-5m}"
+LIGHTTPD_ROOT="${LIGHTTPD_ROOT:-/home/clam/mirror}"
+FRESHCLAM_CONFIG="${FRESHCLAM_CONFIG:-/etc/clamav/freshclam.conf}"
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+}
+
+function log() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@"
+}
+
+
+##############################################
+# Globals:
+#   LIGHTHTTPD_ROOT
+# Arguments:
+#   None
+# Outputs:
+#   Writes update log to ~/.cvdupdate
+##############################################
+function teardown() {
+  log "INFO Cleaning local databases"
+  rm -f /var/lib/clamav/*
+}
+
+##############################################
+# Globals:
+#   FRESHCLAM_CONFIG
+# Arguments:
+#   command
+# Outputs:
+#   Writes log to stdout
+##############################################
+function testfreshclam() {
+  local command
+
+  freshclam --config-file=$FRESHCLAM_CONFIG --stdout $1
+}
+
+
+##############################################
+# Publish signatures 
+# Outputs:
+#   Writes log to stdout
+##############################################
+function publishsigs() {
+  log "INFO Self test PASS: Publishing signatures to mirror"
+  if ! rsync -av --checksum --delete $CVDUPDATE_DEST/ $LIGHTTPD_ROOT/
+    then
+      err "ERROR: Unable to sync mirror"
+      exit
+  fi
+}
+
+##############################################
+# Basic test of eicar and clean file 
+# Outputs:
+#   Writes log to stdout
+##############################################
+function scantest() {
+  log "INFO: Testing for eicar"
+  if ! clamscan files/eicar.txt
```

Comment:
A bit strange that there is a `!` in the condition, considering that there is an 'if' and an 'else' block.
I had to read it twice to make sure I followed the logic.
Just a comment. No need to change if you feel strongly about it...

Reply:
Yeah the command will exit non-zero if a virus is found so is expected to fail in order to "pass", the `!` for error handling is  used quite a bit in a styleguide was trying to follow. I changed the log message below it though!